bool junction_left
bool junction_right 
String junction